include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -37,15 +37,37 @@ class CSummedAreaTableImageFilterBase
 					const auto regions = inputImage->getRegions();
 
 					size_t retval = {};
-					for (const auto* region = regions.begin(); 
-						region < regions.end(); 
-						++region)
-						retval += region->imageExtent.width * region->imageExtent.height * region->imageExtent.depth * inputImage->getCreationParameters().arrayLayers;
+					for (const auto* region = regions.begin(); region != regions.end(); ++region)
+						retval += (region->bufferRowLength ? region->bufferRowLength : region->imageExtent.width) * region->imageExtent.height * region->imageExtent.depth;
 
-					retval *= channels * decodeByteSize;
+					retval *= inputImage->getCreationParameters().arrayLayers * channels * decodeByteSize;
 
 					return retval;
 				}
+
+				inline auto& getImageTotalCacheBuffer()
+				{
+					return imageTotalCacheBuffersOutput;
+				}
```

Comment:
what's this?

Reply:
it contains max sum values for a single plane

Reply:
now it will return the pointer to the scratch memory where the cache is placed in